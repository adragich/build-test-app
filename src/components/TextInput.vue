<template>
  <div class="build-test-u-flex-column build-test-u-spacer">
    <label class="build-test-u-spacer build-test-u-lightgrey-color">
      Kindly asking you to provide your feedback about the content
    </label>
    <textarea
      v-model="innerValue"
      class="build-test__textarea build-test-u-spacer"
      @input="$emit('input', $event.target.value)"
    />
  </div>
</template>

<script>
export default {
  name: 'TextInput',
  props: {
    feedback: {
      type: String,
      required: true
    }
  },
  data: function () {
    return {
      innerValue: this.feedback,
    };
  },
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
  .build-test__textarea {
    border-color: #dfdfdf;
    width: 280px;
    max-width: 100%;
    height: 120px;
    border-radius: 8px;
    box-shadow: 0 6px 4px;
    padding: 8px 16px;
    font-size: 16px;
  }

  @media (max-width: 768px) {
    .build-test__textarea {
      height: 80px;
    }
  }
</style>
