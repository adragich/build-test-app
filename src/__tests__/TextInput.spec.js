import { shallowMount } from '@vue/test-utils';
import TextInput from '../components/TextInput.vue';

describe('TextInput.vue', () => {
  test('renders feedback', () => {
    const feedback = 'Good job!';
    const wrapper = shallowMount(TextInput, {
        propsData: { feedback }
    });
    const textarea = wrapper.find('textarea');
    
    expect(textarea.element.value).toBe(feedback);
  });

  test('emitting "input" event on input', () => {
    const wrapper = shallowMount(TextInput, {
        propsData: { 'feedback': '' }
    });
    let textarea = wrapper.find('textarea');

    textarea.trigger('input', 'I would improve some parts...');
    
    expect(wrapper.emitted('input')).toBeTruthy();
  });

  test('renders textarea item correctly', () => {
    const wrapper = shallowMount(TextInput, {
        propsData: { 'feedback': '' }
    });
    
    expect(wrapper.element).toMatchSnapshot();
  });
});
