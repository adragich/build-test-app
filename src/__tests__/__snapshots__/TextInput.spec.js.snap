// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput.vue renders textarea item correctly 1`] = `
<div
  class="build-test-u-flex-column build-test-u-spacer"
>
  <label
    class="build-test-u-spacer build-test-u-lightgrey-color"
  >
    
    Kindly asking you to provide your feedback about the content
  
  </label>
   
  <textarea
    class="build-test__textarea build-test-u-spacer"
  />
</div>
`;
